ncbi_genome_download/core.py

```diff
@@ -1,74 +1,193 @@
-'''Core functionality of ncbi-genome-download'''
+"""Core functionality of ncbi-genome-download"""
 import errno
 import hashlib
 import logging
 import os
 import sys
-from io import StringIO
 from collections import namedtuple
+from enum import Enum, unique
+from io import StringIO
 from multiprocessing import Pool
-from ncbi_genome_download.summary import SummaryReader
 
 import requests
 
+from ncbi_genome_download.summary import SummaryReader
+
 # Python < 2.7.9 hack: fix ssl support
 if sys.version_info < (2, 7, 9):  # pragma: no cover
     from requests.packages.urllib3.contrib import pyopenssl
-    pyopenssl.inject_into_urllib3()
 
+    pyopenssl.inject_into_urllib3()
 
-NCBI_URI = 'https://ftp.ncbi.nih.gov/genomes'
 SUPPORTED_DOMAINS = ['archaea', 'bacteria', 'fungi', 'invertebrate', 'plant',
                      'protozoa', 'unknown', 'vertebrate_mammalian',
                      'vertebrate_other', 'viral']
 
 
-FORMAT_NAME_MAP = {
-    'genbank': '_genomic.gbff.gz',
-    'fasta': '_genomic.fna.gz',
-    'features': '_feature_table.txt.gz',
-    'gff': '_genomic.gff.gz',
-    'protein-fasta': '_protein.faa.gz',
-    'genpept': '_protein.gpff.gz',
-    'wgs': '_wgsmaster.gbff.gz',
-    'cds-fasta': '_cds_from_genomic.fna.gz',
-    'rna-fasta': '_rna_from_genomic.fna.gz',
-}
-
-ASSEMBLY_LEVEL_MAP = {
-    'complete': 'Complete Genome',
-    'chromosome': 'Chromosome',
-    'scaffold': 'Scaffold',
-    'contig': 'Contig'
-}
-
-
-DownloadJob = namedtuple('DownloadJob', ['full_url', 'local_file', 'expected_checksum', 'symlink_path'])
-
-
-def download(args):
-    '''Download data from NCBI'''
+@unique
+class EMap(Enum):
```

Comment:
Nice use of an Enum here. I'm not used to a bunch of the new Python 3 features yet, but this does make the code easier to read.